src/composables/useActiveMobileColumn.ts

```diff
@@ -0,0 +1,53 @@
+import router, { Columns } from '@/router';
+import { Ref, ref } from '@vue/composition-api';
+
+export enum ColumnType {
+    ADDRESS = 'column-address',
+    ACCOUNT = 'column-account',
+    SIDEBAR = 'column-sidebar',
+}
+
+let activeMobileColumn: Ref<ColumnType> | null = null;
+
+// Adding this code at the end of the call stack so router is not undefined
+setTimeout(() => {
+    router.beforeResolve((to, from, next) => {
+        if (!activeMobileColumn) {
+            activeMobileColumn = ref(ColumnType.ACCOUNT);
+        }
+
+        const { meta, path, query: newQuery } = to;
+        const { query: oldQuery } = from;
+
+        if (newQuery && newQuery.sidebar) {
+            activeMobileColumn.value = ColumnType.SIDEBAR;
+        } else if (oldQuery && oldQuery.sidebar) {
+            activeMobileColumn.value = ColumnType.ACCOUNT;
+        } else if (meta) {
+            switch (meta.column) {
+                case Columns.DYNAMIC:
+                    switch (path) {
+                        case '/': activeMobileColumn.value = ColumnType.ACCOUNT; break;
+                        case '/transactions': activeMobileColumn.value = ColumnType.ADDRESS; break;
+                        default: break; // Don't change column
+                    }
+                    break;
+                case Columns.ACCOUNT: activeMobileColumn.value = ColumnType.ACCOUNT; break;
+                case Columns.ADDRESS: activeMobileColumn.value = ColumnType.ADDRESS; break;
+                default: break;
+            }
+        }
+
+        next();
+    });
+}, 0);
+
+export function useActiveMobileColumn() {
+    if (!activeMobileColumn) {
+        activeMobileColumn = ref(ColumnType.ACCOUNT);
+    }
```

Comment:
Since the variable is initialized here when needed and also in the beforeResolve function above, wouldn't it make sense to directly initialize it to its Ref-value, instead of to null first?

Reply:
The ref function can apparently not be called at that time yet.
cf https://github.com/nimiq/wallet/blob/master/src/composables/useWindowSize.ts#L5-L9

Reply:
Right. Can this comment be copied to here, too, so this is documented?